[BUG] <Баг с входом персонажа в игру rev. от 12.05.2024>

<!-- Remove the space and replace it with a 'x' mark between square [] brackets or click the checkbox after saving to confirm that you agree with the guidelines: -->
<!-- It should look like this: - [x] followed by the text -->
## _I confirm that:_
- [x] I understand that if I do not agree to the following points by marking the checkboxes my issue will be ignored.
- [x] I have searched existing [issues](https://github.com/AAEmu/AAEmu/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated.
- [x] I am currently on the latest commit of the branch when reporting this bug

**Code branch that is affected by this issue** <!-- Change to the branch you are having the bug on -->

`AAEmu-client_version-3.0_client(2017-03-15)+`

**Bug description**

<!-- A more detail description of the bug -->

Проявился баг с входом персонажа в игру в rev. от 12.05.2024, если персонажем войти в игру и совершить действия, затем выйти и подождать некоторое время перед повторным входом, то проявляется баг при котором персонаж входит в игру, но ничего делать внутри игры не может, также выход из игры при этом баге осуществляется мгновенно и без ожидания 10 секунд.

В терминале при загрузке персонажа такая ошибка:

20:55:23 [ERROR] PacketMarshaler - Looks like we got double count 1 

**Steps to reproduce**

<!-- Add steps to reproduce here -->

Зайти персонажем в игру, совершить действия, выйти из игры, подождать некоторое время, попытаться повторно зайти в игру.

**Expected behavior**

<!-- Add expected behaviour here -->

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem -->

https://youtu.be/5dZRm4a2dmg

**Any additional context for this bug**
<!-- Add any other context about the problem here. -->

--- a/AAEmu.Game/Core/Managers/AccountManager.cs
+++ b/AAEmu.Game/Core/Managers/AccountManager.cs
@@ -2,11 +2,13 @@
 using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Linq;
+
 using AAEmu.Commons.Utils;
 using AAEmu.Commons.Utils.DB;
 using AAEmu.Game.Core.Network.Connections;
 using AAEmu.Game.Models;
 using AAEmu.Game.Models.Account;
+
 using NLog;
 
 namespace AAEmu.Game.Core.Managers;
@@ -183,7 +185,7 @@ public class AccountManager : Singleton<AccountManager>
     }
 
     public bool RemoveCredits(ulong accountId, int credits) => AddCredits(accountId, -credits);
-    
+
     public bool AddLoyalty(ulong accountId, int loyaltyAmount)
     {
         object accLock;

--- a/AAEmu.Game/Core/Managers/TimedRewardsManager.cs
+++ b/AAEmu.Game/Core/Managers/TimedRewardsManager.cs
@@ -44,7 +44,7 @@ public class TimedRewardsManager : Singleton<TimedRewardsManager>
             var newLabor = (short)(currentLabor + addLabor);
             AccountManager.Instance.UpdateLabor(connection.AccountId, newLabor);
 
-            connection.SendPacket(new SCCharacterLaborPowerChangedPacket(addLabor, 0, 0, 0));
+            connection.ActiveChar?.SendPacket(new SCCharacterLaborPowerChangedPacket(addLabor, 0, 0, 0));
 
             // Update cache if character was logged in
             connection.ActiveChar?.InitializeLaborCache(newLabor, DateTime.UtcNow);
@@ -76,7 +76,7 @@ public class TimedRewardsManager : Singleton<TimedRewardsManager>
                 // Update Credits
                 AccountManager.Instance.AddCredits(connection.AccountId, AppConfiguration.Instance.Credits.GetTickAmount(connection.Payment.PremiumState));
                 AccountManager.Instance.UpdateTickTimes(connection.AccountId, DateTime.UtcNow, false, true, false);
-                connection.SendPacket(new SCICSCashPointPacket(AccountManager.Instance.GetAccountDetails(connection.AccountId).Credits));
+                connection.ActiveChar?.SendPacket(new SCICSCashPointPacket(AccountManager.Instance.GetAccountDetails(connection.AccountId).Credits));
             }
 
             // Distribute Loyalty if needed
@@ -85,7 +85,7 @@ public class TimedRewardsManager : Singleton<TimedRewardsManager>
                 // Update Loyalty
                 AccountManager.Instance.AddLoyalty(connection.AccountId, AppConfiguration.Instance.Loyalty.GetTickAmount(connection.Payment.PremiumState));
                 AccountManager.Instance.UpdateTickTimes(connection.AccountId, DateTime.UtcNow, false, false, true);
-                connection.SendPacket(new SCBmPointPacket(AccountManager.Instance.GetAccountDetails(connection.AccountId).Loyalty));
+                connection.ActiveChar?.SendPacket(new SCBmPointPacket(AccountManager.Instance.GetAccountDetails(connection.AccountId).Loyalty));
             }
         }
     }

